components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/basicauth/BasicAuthAuthenticator.java

```diff
@@ -203,6 +203,7 @@ public AuthenticationResponse authenticate(MessageContext synCtx) {
             authContext.setCallerToken(null);
             authContext.setApplicationName(null);
             authContext.setApplicationId(clientIP); //Set clientIp as application ID in unauthenticated scenario
+            authContext.setApplicationUUID(clientIP); //Set clientIp as application ID in unauthenticated scenario
```

Comment:
This could result in a large combination of groups in the streaming layer. So I prefer to have a predefined string for this.

Reply:
set application UUID same as ID because this goes to auth context. But when publishing to analytics, we check the username and set all attribute to anonymous
```
if (APIConstants.END_USER_ANONYMOUS.equalsIgnoreCase(authContext.getUsername())) {
       this.setAnonymousApp(responseEvent);
 }
```

Reply:
ack